snapcraft/internal/lifecycle.py

```diff
@@ -320,7 +320,8 @@ def _clean_part_and_all_dependents(part_name, step, config, staged_state,
         dependent_part.clean(staged_state, primed_state, step)
 
     # Finally, clean the part in question
-    config.get_part(part_name).clean(staged_state, primed_state, step)
+    config.parts_config.get_part(part_name).clean(staged_state, primed_state,
+                                                  step)
```

Comment:
This long chain of calls is a little worrying [1]. 
Is it right for the part to have the responsibility to clean itself?
If so, we can at least hide it making a clean_part method in parts_config.
Also this made me notice that the _Executor doesn't have a lot of use for config, only for parts_config. Why don't we store parts_config as an attribute?

[1] http://c2.com/cgi/wiki?TrainWreck